lib/response-handler.js

```diff
@@ -86,9 +86,18 @@ module.exports = (bluebird, error, logger, parseLinkHeader, URI) => {
             logger.debug(`Got link attribute: ${response.headers.link}`);
             links = parseLinkHeader(response.headers.link);
             logger.debug(`Parsed link header: ${JSON.stringify(links, null, 4)}`);
-            links.item.forEach((val) => {
-                urls.push(val.url);
-            });
+
+            // If we received more than one link header item, push them
+            // all onto an array. If we received just one, push just that
+            // one onto the array.
+            if (Array.isArray(links.item)) {
+                links.item.forEach((val) => {
+                    urls.push(val.url);
+                });
+            } else {
+                urls.push(links.url);
+            }
+
```

Comment:
This won't work for a single link, you want this:
```
urls.push(links.item.url);
```
If it is a single link, should we just return a string? So it returns an array for multiple or string for single? What do you think?